nixos/modules/services/networking/nftables.nix

```diff
@@ -82,18 +82,27 @@ in
         '';
     };
     networking.nftables.rulesetFile = mkOption {
-      type = types.path;
-      default = pkgs.writeTextFile {
-        name = "nftables-rules";
-        text = cfg.ruleset;
-      };
-      defaultText = literalMD ''a file with the contents of {option}`networking.nftables.ruleset`'';
+      type = types.nullOr types.path;
+      default = null;
       description =
         lib.mdDoc ''
           The ruleset file to be used with nftables.  Should be in a format that
           can be loaded using "nft -f".  The ruleset is updated atomically.
         '';
     };
+    networking.nftables.stopRuleset = mkOption {
+      type = types.lines;
+      default = "flush ruleset";
+      example = ''
+        table inet nixos_fw
```

Comment:
For reasons which I sadly don't remember, I decided not to do it like this in my downstream firewall module, but instead check in a shell script if these tables exist and only run `delete`.

Maybe… there was/is an issue with having a table definition twice in the same `nft -f` script?

Could you maybe add a simple nixos test for this or add it to an existing one?

Reply:
It should not be an issue according to https://unix.stackexchange.com/questions/537030/nftables-flush-delete-when-changing-or-creating-new-table. Maybe it's because of a too old kernel.

Anyway, I'll add a test.